### ISSUE TITLE
Bug: SQLSRV driver does not work correctly with schemas and migration

### ISSUE THREAD

**Issue by qury** [codeigniter4/CodeIgniter4#4853]
**Describe the bug**
Migrations do not work with SQLSRV driver if the schema is anything but the default 'dbo'.

CodeIgniter 4 version: 4.1.3

**Affected module(s)**
CodeIgniter\Database\BaseConnection -> when table() is initialized it does not take into consideration the schema
CodeIgniter\Database\SQLSRV ->  $createTableIfStr  does not take into consideration schemas

**Expected behavior, and steps to reproduce if appropriate**

1. create a database for test and create a new schema called 'test'
2. configure CI4 to use 'dbo' as schema
3. create a migration file for any kind of table and run it against 'dbo' schema
4. change the schema to 'test' and run the migration again

Expectation is to have the same table created in both dbo and test schema.
Actual behaviour: table is not created in 'test' schema.

**Context**
 - OS: Linux
 - Web server : n/a -> CLI
 - PHP version 7.4.20

**Comment by wradstok** [comment]
The SQLSRV driver in itself does not account for multiple schemas, entirely, as the method`getFullName` always append the default schema. If none is set (i.e., `$this-db->schema == ""`  , the output looks like: `DatabaseName..TableName`, which is of course not a valid name!

I have opened a pull request that allows you to override the `getFullName` method with your own by changing the access modifier #4858. This is what we are doing now to allow for multiple schema's. However, seeing as probably more people are affected by this issue, perhaps the method itself should be modified?

I would suggest something checking for something like:

```
if (empty($this->db->schema)) {
	if (!str_contains($table, '.')) {
		throw new UnexpectedValueException("Database schema for table ${table} was not specified.");
	}
	return '"' . $this->db->getDatabase() . '"."' . str_replace('"', '', $table) . '"' . $alias;
}
```

**Comment by qury** [comment]
Unfortunately there is a host of other places where the schema will need to be added.
Check my work in progress fix here: https://github.com/qury/CodeIgniter4/tree/fix/slqsrvSchema-4853

I've started working through any errors that comes up from  "tests/system/Database/Live/" scripts:
Current progress: _Tests: 225, Assertions: 451, Errors: 2, Failures: 6._

_The above is with using 'test' for schema instead of the default dbo._

**Comment by michalsn** [comment]
Resolved via #4903
